miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -352,12 +356,65 @@ private long saveNewBox(Box box) throws IOException {
       }
       box.setName(namingScheme.generateNameFor(box));
       validateNameOrThrow(box, namingScheme);
+
+      if (box.getStorageLocation() != null) {
+        addStorageChangeLog(box.getStorageLocation(), box, true);
+      }
+
       return boxStore.save(box);
     } catch (MisoNamingException e) {
       throw new IOException("Invalid name for box", e);
     }
   }
 
+  private void logStorageChange(Box box, Box original) throws IOException {
+    if (original.getStorageLocation() == null && box.getStorageLocation() == null) {
+      return;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this redundant jump. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3626)